### PR TITLE
Update PVGIS TMY tests and files

### LINKED ISSUE #2221: PVGIS tests are failing
Several PVGIS tests are failing because the expected values no longer match what is returned by the API.  See an example CI run: https://github.com/pvlib/pvlib-python/actions/runs/11003008118/job/30551180154#step:6:256

```
=========================== short test summary info ============================
FAILED pvlib/tests/iotools/test_pvgis.py::test_get_pvgis_tmy - assert False
 +  where False = <function allclose at 0x7fbb5aafbf70>(time(UTC)\n2014-01-01 00:00:00+00:00    1.08\n2014-01-01 01:00:00+00:00    0.42\n2014-01-01 02:00:00+00:00   -0.24\n2014-0...:00    3.05\n2018-12-31 22:00:00+00:00    2.39\n2018-12-31 23:00:00+00:00    1.73\nName: T2m, Length: 8760, dtype: float64, time(UTC)\n2009-01-01 00:00:00+00:00    2.29\n2009-01-01 01:00:00+00:00    2.10\n2009-01-01 02:00:00+00:00    1.91\n2009-0...:00    2.87\n2011-12-31 22:00:00+00:00    2.67\n2011-12-31 23:00:00+00:00    2.48\nName: T2m, Length: 8760, dtype: float64)
 +    where <function allclose at 0x7fbb5aafbf70> = np.allclose
FAILED pvlib/tests/iotools/test_pvgis.py::test_get_pvgis_tmy_kwargs - assert False
 +  where False = <function allclose at 0x7fbb5aafbf70>(time(UTC)\n2014-01-01 00:00:00+00:00    0.0\n2014-01-01 01:00:00+00:00    0.0\n2014-01-01 02:00:00+00:00    0.0\n2014-01-0...00:00    0.0\n2018-12-31 22:00:00+00:00    0.0\n2018-12-31 23:00:00+00:00    0.0\nName: G(h), Length: 8760, dtype: float64, array([0., 0., 0., ..., 0., 0., 0.]))
 +    where <function allclose at 0x7fbb5aafbf70> = np.allclose
 +    and   array([0., 0., 0., ..., 0., 0., 0.]) = 0       0.0\n1       0.0\n2       0.0\n3       0.0\n4       0.0\n       ... \n8755    0.0\n8756    0.0\n8757    0.0\n8758    0.0\n8759    0.0\nName: G(h), Length: 8760, dtype: float64.values
FAILED pvlib/tests/iotools/test_pvgis.py::test_get_pvgis_tmy_basic - assert False
 +  where False = <function allclose at 0x7fbb5aafbf70>(time(UTC)\n2014-01-01 00:00:00+00:00    1.08\n2014-01-01 01:00:00+00:00    0.42\n2014-01-01 02:00:00+00:00   -0.24\n2014-0...:00    3.05\n2018-12-31 22:00:00+00:00    2.39\n2018-12-31 23:00:00+00:00    1.73\nName: T2m, Length: 8760, dtype: float64, time(UTC)\n2009-01-01 00:00:00+00:00    2.29\n2009-01-01 01:00:00+00:00    2.10\n2009-01-01 02:00:00+00:00    1.91\n2009-0...:00    2.87\n2011-12-31 22:00:00+00:00    2.67\n2011-12-31 23:00:00+00:00    2.48\nName: T2m, Length: 8760, dtype: float64)
 +    where <function allclose at 0x7fbb5aafbf70> = np.allclose
FAILED pvlib/tests/iotools/test_pvgis.py::test_get_pvgis_tmy_csv - assert False
 +  where False = <function allclose at 0x7fbb5aafbf70>(time(UTC)\n2014-01-01 00:00:00+00:00    1.08\n2014-01-01 01:00:00+00:00    0.42\n2014-01-01 02:00:00+00:00   -0.24\n2014-0...:00    3.05\n2018-12-31 22:00:00+00:00    2.39\n2018-12-31 23:00:00+00:00    1.73\nName: T2m, Length: 8760, dtype: float64, time(UTC)\n2009-01-01 00:00:00+00:00    2.29\n2009-01-01 01:00:00+00:00    2.10\n2009-01-01 02:00:00+00:00    1.91\n2009-0...:00    2.87\n2011-12-31 22:00:00+00:00    2.67\n2011-12-31 23:00:00+00:00    2.48\nName: T2m, Length: 8760, dtype: float64)
 +    where <function allclose at 0x7fbb5aafbf70> = np.allclose
FAILED pvlib/tests/iotools/test_pvgis.py::test_get_pvgis_tmy_epw - assert False
 +  where False = <function allclose at 0x7fbb5aafbf70>(2014-01-01 00:00:00+01:00    0.0\n2014-01-01 01:00:00+01:00    0.0\n2014-01-01 02:00:00+01:00    0.0\n2014-01-01 03:00:00...+01:00    0.0\n2018-12-31 22:00:00+01:00    0.0\n2018-12-31 23:00:00+01:00    0.0\nName: ghi, Length: 8760, dtype: float64, time(UTC)\n2009-01-01 00:00:00+00:00    0.0\n2009-01-01 01:00:00+00:00    0.0\n2009-01-01 02:00:00+00:00    0.0\n2009-01-0...00:00    0.0\n2011-12-31 22:00:00+00:00    0.0\n2011-12-31 23:00:00+00:00    0.0\nName: G(h), Length: 8760, dtype: float64)
 +    where <function allclose at 0x7fbb5aafbf70> = np.allclose
 +    and   2014-01-01 00:00:00+01:00    0.0\n2014-01-01 01:00:00+01:00    0.0\n2014-01-01 02:00:00+01:00    0.0\n2014-01-01 03:00:00...+01:00    0.0\n2018-12-31 22:00:00+01:00    0.0\n2018-12-31 23:00:00+01:00    0.0\nName: ghi, Length: 8760, dtype: float64 =                            year  ...  liquid_precipitation_quantity\n2014-01-01 00:00:00+01:00  2014  ...              ...                       99\n2018-12-31 23:00:00+01:00  2018  ...                             99\n\n[8760 rows x 35 columns].ghi
======= 5 failed, 138 passed, 8 warnings, 25 rerun in 253.54s (0:04:13) ========
```

I suppose it has something to do with the release of PVGIS 5.3.  See here:
https://joint-research-centre.ec.europa.eu/photovoltaic-geographical-information-system-pvgis/pvgis-releases/pvgis-53_en

### PR DESCRIPTION
**Pull request by AdamRJensen**
- [x] Closes #2221
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

The default API endpoint for PVGIS has changed from 5.1 to 5.2, which has caused the tests for pvgis.iotools.get_pvgis_tmy. The *_pvgis_hourly tests are not affected as these specify the specific endpoint (5.2).

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.11.1.rst [hunk 1]
```diff
@@ -55,6 +55,9 @@ Bug fixes
 
 Testing
 ~~~~~~~
+* Updated the tests for :py:func:`~pvlib.iotools.get_pvgis_hourly`
+  and :py:func:`~pvlib.iotools.read_pvgis_hourly` as the default
+  API endpoint has changed from 5.1 to 5.2. (:pull:`2222`)
```
Comment (kandersolar) [inline docs/sphinx/source/whatsnew/v0.11.1.rst:60]: Isn't it the TMY functions?

**Review by kandersolar — APPROVED** [review]
Looks OK to me, although I didn't inspect the contents of any data files.

File: docs/sphinx/source/whatsnew/v0.11.1.rst [hunk 2]
```diff
@@ -55,6 +55,9 @@ Bug fixes
 
 Testing
 ~~~~~~~
+* Updated the tests for :py:func:`~pvlib.iotools.get_pvgis_hourly`
+  and :py:func:`~pvlib.iotools.read_pvgis_hourly` as the default
```
Comment (AdamRJensen) [inline docs/sphinx/source/whatsnew/v0.11.1.rst:59]:
```suggestion
* Updated the tests for :py:func:`~pvlib.iotools.get_pvgis_tmy`
  and :py:func:`~pvlib.iotools.read_pvgis_tmy` as the default
```

**Comment by kandersolar** [comment]
Thanks @AdamRJensen :)

**Pull request merged**